src/util/reader_mapping.hpp

```diff
@@ -56,6 +56,7 @@ class ReaderMapping final
 
   bool get(const char* key, sexp::Value& value) const;
 
+  bool has_item(const char* key) const;
```

Comment:
If I recall correctly, there is already a standard way to handle optional attribute. But using this instead doesn't mind me at all because it looks more straightforward.